api/client.go

```diff
@@ -630,6 +761,35 @@ func (c *Client) Clone() (*Client, error) {
 	return NewClient(newConfig)
 }
 
+// config.Clone returns a clone of the config it is called on
+func (c *Config) Clone() *Config {
+	c.modifyLock.RLock()
+
+	newConfig := &Config{
+		Address:           c.Address,
+		AgentAddress:      c.AgentAddress,
+		HttpClient:        c.HttpClient,
```

Comment:
Should we also deep copy HttpClient?  Might not be necessary, but it does look like Jar is stateful...

Reply:
unsure. Aside from Jar, are there other parts of HttpClient that would need to be deep copied as well? @briankassouf any thoughts?